pbench/agent/__init__.py

```diff
@@ -1,7 +1,91 @@
-import logging
-import sys
+from pathlib import Path
+from configparser import NoSectionError, NoOptionError
 
-logger = logging.getLogger(__name__)
-logger.addHandler(logging.StreamHandler(sys.stdout))
-logger.propagate = False
-logging.basicConfig(level=logging.INFO, format="%(message)s")
+from pbench import PbenchConfig
+from pbench.common.exceptions import BadConfig
+from pbench.common.constants import (
+    DEFAULT_PBENCH_AGENT_RUN_DIR,
+    DEFAULT_PBENCH_AGENT_INSTALL_DIR,
+    DEFAULT_SCP_OPTS,
+    DEFAULT_SSH_OPTS,
+)
+
+
+class PbenchAgentConfig(PbenchConfig):
+    """PbenchAgentConfig - a sub-class of the PbenchConfig class specifically
+    for the pbench agent environment.
+    """
+
+    def __init__(self, cfg_name):
+        super().__init__(cfg_name)
+
+        try:
+            # Provide a few convenience attributes.
+            self.agent = self.conf["pbench-agent"]
+            self.results = self.conf["results"]
+            # Now fetch some default common pbench settings that are required.
```

Comment:
Sure, changing this to use `@property` here would be good to do, but not sure there is really a gain with efficiency because the config class is not long lived in agent code with lots of accesses.

For the `pbench-config` use case it of pulling one properly out of a config file, we should address that efficiency in other ways.  One of which could be to export all commonly used configuration values as environment variables once in `agent/base`, another would be to ignore this until the python re-write completes.